app/models/application_record.rb

```diff
@@ -36,13 +36,13 @@ def self.where_in_interval(attribute, interval)
     result = self
 
     if interval.start
-      comparison = interval.start_open ? '>' : '>='
-      result = result.where("#{attribute} #{comparison} ?", interval.start)
+      comparison = interval.start_open ? :gt : :gteq
+      result = result.where(arel_table[attribute].send(comparison, interval.start))
     end
 
     if interval.end
-      comparison = interval.end_open ? '<' : '<='
-      result = result.where("#{attribute} #{comparison} ?", interval.end)
+      comparison = interval.end_open ? :lt : :lteq
+      result = result.where(arel_table[attribute].send(comparison, interval.end))
```

Comment:
In 7.2, Active Record started raising `PG::InvalidTextRepresentation` errors for the previous string-based `where` clause (because the interval start/end are floats and the column in question is an integer). This seems to be the same as https://github.com/rails/rails/issues/51617, caused by https://github.com/rails/rails/pull/51139.

This isn’t really a problem in practice for Postgres, and we can avoid whatever check is raising this by using Arel instead, which I did here. Another approach would be to check the column type (e.g. `columns_hash[attribute.to_s].type`) and cast `interval.start`/`interval.end` to an int before querying if the column type is `:integer`. Either works.